pyanaconda/ui/tui/simpleline/widgets.py

```diff
@@ -19,12 +19,222 @@
 # Red Hat, Inc.
 #
 
-__all__ = ["TextWidget", "ColumnWidget", "CheckboxWidget", "CenterWidget"]
+__all__ = ["Widget", "TextWidget", "ColumnWidget", "CheckboxWidget", "CenterWidget"]
 
+import functools
 from pyanaconda.i18n import _
-from pyanaconda.ui.tui.simpleline import base
+from textwrap import wrap
+from pyanaconda import iutil
 
-class TextWidget(base.Widget):
+class Widget(object):
+    def __init__(self, max_width=None, default=None):
+        """Initializes base Widgets buffer.
+
+        :param max_width: server as a hint about screen size to write method with default arguments
+        :type max_width: int
+
+        :param default: string containing the default content to fill the buffer with
+        :type default: string
+        """
+
+        self._buffer = []
+        if default:
+            self._buffer = [[c for c in l] for l in default.split("\n")]
+        self._max_width = max_width
+        self._cursor = (0, 0)  # row, col
+
+    @property
+    def height(self):
+        """The current height of the internal buffer."""
+        return len(self._buffer)
+
+    @property
+    def width(self):
+        """The current width of the internal buffer (id of the first empty column)."""
+        return functools.reduce(lambda acc, l: max(acc, len(l)), self._buffer, 0)
+
+    def clear(self):
+        """Clears this widgets buffer and resets cursor."""
+        self._buffer = list()
+        self._cursor = (0, 0)
+
+    @property
+    def content(self):
+        """This has to return list (rows) of lists (columns) with one character elements."""
```

Comment:
I think it would be better, after all you are the one who will be blamed by Git now :).